plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestIssue14317.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.sql.query.QueryAssertions;
+import io.trino.testing.DistributedQueryRunner;
+import org.junit.jupiter.api.TestInstance;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+/**
+ * Test for https://github.com/trinodb/trino/issues/14317
+ * <p>
+ * The only known reproduction involves a connector that exposes partitioned tables
+ * and supports predicate pushdown.
+ */
+@TestInstance(PER_CLASS)
+public class TestIssue14317
+{
+    @Test
+    public void test()
+            throws Exception
+    {
+        HiveQueryRunner.Builder<?> builder = HiveQueryRunner.builder()
+                .setCreateTpchSchemas(false);
+
+        try (DistributedQueryRunner queryRunner = builder.build();
+                QueryAssertions assertions = new QueryAssertions(queryRunner);) {
+            queryRunner.execute("CREATE SCHEMA s");
+
+            queryRunner.execute("CREATE TABLE s.t (a bigint, b bigint)");
+            queryRunner.execute("CREATE TABLE s.u (c bigint, d bigint) WITH (partitioned_by = array['d'])");
```

Comment:
so this is required so that table derived predicate can be obtained?

Reply:
correct